### PR TITLE
Wrap debug messages with a conditional compilation check.

### PR DESCRIPTION
**Pull request by microbit-carlos**
Very minor update so that the bigger blocks with `if` statements only to print debug info are 100% not compiled when `ML_DEBUG_PRINT` is disabled (the compiler should take care of that, but this little extra check doesn't cost anything).

No rush, can sit here until you have to for review.

### CODE REVIEW

File: autogenerated.ts [hunk 1]
```diff
@@ -37,6 +37,8 @@ namespace ml {
...
     return result;
   };
+
+  simulatorSendData();
```
Comment (microbit-carlos) [inline autogenerated.ts]: Also added this to the default `autogenerated.ts`, assuming it is needed, but not really tested it with the simulator itself.
Happy to remove it if it's better without it.

Comment (microbit-robert) [reply autogenerated.ts]: We don't need this line. The `simulatorRegister()` call kicks things off. When the simulator is rendered it asks this extension for the data, so `simulatorSendData()` is only triggered in response.

Comment (microbit-carlos) [reply autogenerated.ts]: I think this line might still be generated in the `autogenerated.ts` file (or at least the files generated in the `ml4f-test-header` branch), should it be removed from there as well?

**Review by microbit-robert — CHANGES_REQUESTED** [review]
Let's remove the added `simulatorSendData();` line.

The other changes look good.

File: autogenerated.ts (hunk 1 above)
Comment (microbit-robert) [reply autogenerated.ts]: Yes, good catch. Needs testing after the change, but I'm confident that's the right thing to do.

**Review by microbit-robert — APPROVED** [review]
Thanks. LGTM.

**Pull request merged**